beacon-chain/rpc/endpoints.go

```diff
@@ -34,18 +34,39 @@ type endpoint struct {
 	methods    []string
 }
 
+// responseWriter is the wrapper to http Response writer to capture status code.
+type responseWriter struct {
+	http.ResponseWriter
+	statusCode int
+}
+
+func (w *responseWriter) WriteHeader(statusCode int) {
+	w.statusCode = statusCode
+	w.ResponseWriter.WriteHeader(statusCode)
+}
```

Comment:
Also a nit but since this is exported and doesn't just write to header but is actually writing the status to header, it may be better to have a comment about it or consider renaming

Reply:
The responseWriter struct basically wraps the `WriteHeader` method of internal ResponseWriter interface which has `WriteHeader` to capture status code. The `WriteHeader` method is used for writing status code as header.
Refer godoc of `WriteHeader` method: https://pkg.go.dev/net/http@go1.23.3#ResponseWriter.

I'll add a comment for more clarity.